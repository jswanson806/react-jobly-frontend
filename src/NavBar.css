.navbar .container-fluid {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: #c5dacc;
    font-family: 'Raleway', sans-serif;
    font-weight: bold;
  }
  
  .navbar .navbar-nav {
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;
  }
  
  .navbar .navbar-nav li {
    margin-right: 20px;
  }
  
  .navbar .navbar-nav li:last-child {
    margin-right: 0;
  }
  
  .navbar .nav-link {
    color: #0f5959;
    text-decoration: none;
  }

  .navbar .navbar-brand {
    color: #0f5959;
    text-decoration: none;
  }
  
  .navbar .nav-link.active {
    color: #f1bb7b;
  }

  .navbar .nav-link-button {
    background: none;
    border: none;
    font-family: 'Raleway', sans-serif;
    font-weight: bold;
    font-size: small;
    color: #0f5959;
    text-decoration: none;
    cursor: pointer;
  }