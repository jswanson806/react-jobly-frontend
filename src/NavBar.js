import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from 'reactstrap';
import AppContext from "./helpers/AppContext.js";
import './styles/NavBar.css';

/**
 * NavBar Component
 *
 * Renders the navigation bar with links to different pages based on the user's login status.
 * Requires the `AppContext` to be provided by a parent component.
 */


const NavBar = () => {
    const {logOut} = useContext(AppContext);
    const {user} = useContext(AppContext);
    const {token} = useContext(AppContext);

    // token is set to "" by default and when logOut() is called
    const loggedIn = token !== "" ? true : false

    return(
        <Navbar expand="md" className="navbar" data-testid="Navbar">
            <NavLink exact to="/" className="navbar-brand" data-testid="brand-link">
              Jobly
            </NavLink>

            <Nav className="navbar-nav">
              <NavItem>
                <NavLink to="/companies" className="nav-link" data-testid="companies-link">
                  Companies
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink to="/jobs" className="nav-link" data-testid="jobs-link">
                  Jobs
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink to="/profile" className="nav-link" data-testid="profile-link">
                  Profile
                </NavLink>
              </NavItem>
              {/* conditionally render following based on logged in status of user */}
              {loggedIn && (
                <NavItem>
                  <button onClick={() => logOut()} className="nav-link-button" data-testid="logout-button">
                    Logout {`${user.username}`}
                  </button>
                </NavItem>
              )}
              {loggedIn === false && (
                <NavItem>
                  <NavLink to="/signup" className="nav-link" data-testid="signup-link">
                    Signup
                  </NavLink>
                </NavItem>
              )}
              {loggedIn === false && (
                <NavItem>
                  <NavLink to="/login" className="nav-link" data-testid="login-link">
                    Login
                  </NavLink>
                </NavItem>
              )}
            </Nav>
        </Navbar>

    );
};

export default NavBar;