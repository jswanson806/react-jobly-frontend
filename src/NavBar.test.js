import React from "react";
import {render, screen, fireEvent, waitFor} from '@testing-library/react';
import NavBar from './NavBar.js';
import AppContext from "./helpers/AppContext.js";
import { Router, MemoryRouter } from "react-router-dom";
import {createMemoryHistory} from 'history';
import '@testing-library/jest-dom';

const logOut = jest.fn();

const history = createMemoryHistory();

let token;
let user;

beforeEach(() => {
    token = 'abcdefg';
})

user = {
    username: "testuser",
    password: "password",
    firstName: "firstName",
    lastName: "lastName",
    email: "test@email.com"
}

describe('NavBar smoke and snapshot', () => {

    it('should render without crashing', async () => {

        render(
            <AppContext.Provider value={{token, user, logOut}}>
                <MemoryRouter>
                    <NavBar/>
                </MemoryRouter>
            </AppContext.Provider>
        )
    });

    it('should match the snapshot', async () => {

        const {asFragment} = render(
            <AppContext.Provider value={{token, user, logOut}}>
                <MemoryRouter>
                    <NavBar/>
                </MemoryRouter>
            </AppContext.Provider>
        )
        expect(asFragment()).toMatchSnapshot();
    });

})

describe('NavBar handles navigation correctly', () => {
 
    it('should render correct links if user is logged in', async () => {
        
        render(
            <AppContext.Provider value={{token, user, logOut}}>
                <Router history={history}>
                    <NavBar/>
                </Router>
            </AppContext.Provider>
        )

        const logoutBtn = screen.getByTestId('logout-button')
        
        expect(logoutBtn).toBeInTheDocument();
        expect(screen.queryByTestId('signup-link')).toBe(null);
        expect(screen.queryByTestId('login-link')).toBe(null);

    });

    it('should render correct links if user is logged out', async () => {
        
        token = '';

        render(
            <AppContext.Provider value={{token, user, logOut}}>
                <Router history={history}>
                    <NavBar/>
                </Router>
            </AppContext.Provider>
        )


        const signup = screen.getByTestId('signup-link')
        const loginLink = screen.getByTestId('login-link')

        expect(signup).toBeInTheDocument();
        expect(loginLink).toBeInTheDocument();
        expect(screen.queryByTestId('logout-button')).toBe(null);

    });

    it('should navigate to correct routes', async () => {
        
        render(
            <AppContext.Provider value={{token, user, logOut}}>
                <Router history={history}>
                    <NavBar/>
                </Router>
            </AppContext.Provider>
        )

        const logoutBtn = screen.getByTestId('logout-button')
        const companiesLink = screen.getByTestId('companies-link')
        const jobsLink = screen.getByTestId('jobs-link')
        const profileLink = screen.getByTestId('profile-link')
        const brandLink = screen.getByTestId('brand-link')

        fireEvent.click(companiesLink);
        expect(history.location.pathname).toBe('/companies');
        fireEvent.click(jobsLink);
        expect(history.location.pathname).toBe('/jobs');
        fireEvent.click(profileLink);
        expect(history.location.pathname).toBe('/profile');
        fireEvent.click(brandLink);
        expect(history.location.pathname).toBe('/');
        fireEvent.click(logoutBtn);
        expect(history.location.pathname).toBe('/');

    });

    it('should navigate to correct routes', async () => {

        token = '';

        render(
            <AppContext.Provider value={{token, user, logOut}}>
                <Router history={history}>
                    <NavBar/>
                </Router>
            </AppContext.Provider>
        )

        const signup = screen.getByTestId('signup-link')
        const loginLink = screen.getByTestId('login-link')

        fireEvent.click(signup);
        expect(history.location.pathname).toBe('/signup');
        fireEvent.click(loginLink);
        expect(history.location.pathname).toBe('/login');
    });

})
