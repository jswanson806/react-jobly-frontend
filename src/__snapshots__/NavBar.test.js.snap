// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar smoke and snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar navbar-expand-md"
    data-testid="Navbar"
  >
    <div
      class="container-fluid"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        data-testid="brand-link"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="navbar-nav nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="companies-link"
            href="/companies"
          >
            Companies
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="jobs-link"
            href="/jobs"
          >
            Jobs
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="profile-link"
            href="/profile"
          >
            Profile
          </a>
        </li>
        <li
          class="nav-item"
        >
          <button
            class="nav-link-button"
            data-testid="logout-button"
          >
            Logout testuser
          </button>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
